lib/galaxy/datatypes/images.py

```diff
@@ -89,27 +89,24 @@ class OMETiff(Tiff):
     MetadataElement(name="offsets", desc="Offsets File", param=FileParameter, file_ext="json", readonly=True, no_value=None, visible=False, optional=True)
 
     def set_meta(self, dataset, overwrite=True, **kwd):
-        try:
-            spec_key = 'offsets'
-            offsets_file = dataset.metadata.offsets
-            if not offsets_file:
-                offsets_file = dataset.metadata.spec[spec_key].param.new_file(dataset=dataset)
-            with tifffile.TiffFile(dataset.file_name) as tif:
-                offsets = [page.offset for page in tif.pages]
-            with open(offsets_file.file_name, 'w') as f:
-                json.dump(offsets, f)
-            dataset.metadata.offsets = offsets_file
-        except Exception:
-            pass
+        spec_key = 'offsets'
+        offsets_file = dataset.metadata.offsets
+        if not offsets_file:
+            offsets_file = dataset.metadata.spec[spec_key].param.new_file(dataset=dataset)
+        with tifffile.TiffFile(dataset.file_name) as tif:
+            offsets = [page.offset for page in tif.pages]
+        with open(offsets_file.file_name, 'w') as f:
+            json.dump(offsets, f)
+        dataset.metadata.offsets = offsets_file
 
     def sniff(self, filename):
-        is_tiff = Tiff().sniff(filename)
-        if not is_tiff:
+        try:
+            with tifffile.TiffFile(filename) as tif:
+                if tif.is_ome:
+                    return True
+                return False
+        except tifffile.TiffFileError:
```

Comment:
@mvdbeek  Thanks for your great comments.
The sniff method has been revised. Catch either `tifffile.TiffFileError` error or just catch a general `Exception`, which one do you think is better here?